dashboard-notifications/src/main/java/uk/gov/hmcts/reform/dashboard/services/DashboardNotificationService.java

```diff
@@ -80,9 +80,14 @@ public void recordClick(UUID id, String authToken) {
                 .dashboardNotification(notification)
                 .actionPerformed("Click")
                 .createdBy(idamApi.retrieveUserDetails(authToken).getFullName())
+                .createdAt(OffsetDateTime.now())
                 .build();
-
-            notificationActionRepository.save(notificationAction);
+            if (nonNull(notification.getNotificationAction())
+                && notification.getNotificationAction().getActionPerformed().equals("Click")) {
+                notificationAction.setId(notification.getNotificationAction().getId());
+            }
+            notification.setNotificationAction(notificationAction);
```

Comment:
So this is a slightly different way of saving it - did you discuss this with Dharmendra?

Reply:
Earlier the code was just adding new entries to the table but now that we are also updating the existing row if an entry is present already, if we use the notification action repository, it is throwing an error Multiple representations of the same entity [[uk.gov](http://uk.gov/).hmcts.reform.dashboard.entities.NotificationActionEntity#49] are being merged.  The reason being, we are fetching the data using the notification repository and directly updating using action repository. I think hibernate maintains a state using hashcode internally. Now we are saving it using the notification repository and it cascades/updates the action repository